src/test/ui/issues/issue-72278.rs

```diff
@@ -0,0 +1,19 @@
+// run-pass
+
+#![allow(unused)]
+
+struct S;
+
+impl S {
+    fn func<'a, U>(&'a self) -> U {
+        todo!()
+    }
+}
+
+fn dont_crash<'a, U>() -> U {
+    S.func::<'a, U>()
+    //~^ WARN cannot specify lifetime arguments explicitly
+    //~| WARN this was previously accepted
+}
```

Comment:
Do we have tests for all the other random scenarios that can occur? It seems like we should try to test them all -- in particular the case of "got a lifetime but expected something else"...

Reply:
I was under the impression that

- [src/test/ui/methods/method-call-lifetime-args.rs](https://github.com/rust-lang/rust/blob/master/src/test/ui/methods/method-call-lifetime-args.rs)
- [src/test/ui/methods/method-call-lifetime-args-fail.rs](https://github.com/rust-lang/rust/blob/master/src/test/ui/methods/method-call-lifetime-args-fail.rs)

cover pretty much everything but I could be wrong

Reply:
Those tests look pretty good. I guess they don't cover lifetimes + types intermixed. But I'm satisfied.